### PR TITLE
feat: expose methods to examine OutputUnits

### PR DESCRIPTION
**Pull request by alexjg**
The `Output::basic` method returns a BasicOutput. Prior to this patch the
only way to examine the contents of this data structure was to serialize
it to JSON. This patch adds methods to examine the outputs from Rust, as
well as documentation on how to do so.

Signed-off-by: Alex Good <alex@memoryandthought.me>

### CODE REVIEW

File: jsonschema/src/output.rs [hunk 1]
```diff
@@ -169,11 +231,56 @@ impl<T> OutputUnit<T> {
...
         }
     }
+
+    ///  The location in the schema of the keyword
+    pub const fn keyword_location(&self) -> &JSONPointer {
+        &self.keyword_location
+    }
+
+    ///  The absolute location in the schema of the keyword. This will be
+    ///  different to `keyword_location` if the schema is a resovled reference.
```
Comment (Stranger6667) [inline jsonschema/src/output.rs:241]:
```suggestion
    ///  different to `keyword_location` if the schema is a resolved reference.
```

File: jsonschema/src/output.rs [hunk 2]
```diff
@@ -169,11 +231,56 @@ impl<T> OutputUnit<T> {
...
 pub struct Annotations<'a>(AnnotationsInner<'a>);
 
+impl<'a> Annotations<'a> {
+    /// The `serde_json::Value` of the annotation
+    pub fn value(&'a self) -> Cow<'a, serde_json::Value> {
+        match &self.0 {
+            AnnotationsInner::Value(v) => Cow::Borrowed(v),
+            AnnotationsInner::ValueRef(v) => Cow::Borrowed(v),
+            AnnotationsInner::UnmatchedKeywords(kvs) => {
+                let value = serde_json::to_value(kvs).unwrap();
```
Comment (Stranger6667) [inline jsonschema/src/output.rs:277]: Could we use `expect` here instead? Not sure what would be the situation when panic may happen

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
Thank you, that's awesome! I have two minor comments :)

File: jsonschema/src/output.rs (hunk 1 above)
Comment (alexjg) [reply jsonschema/src/output.rs]: Good spot, fixed.

File: jsonschema/src/output.rs (hunk 2 above)
Comment (alexjg) [reply jsonschema/src/output.rs]: I've added an `expect`. The error cases mentioned in `serde_json::to_value` are:

> This conversion can fail if `T`'s implementation of `Serialize` decides to fail, or if `T` contains a map with non-string keys.

But `T` in  our case is `AHashMap<String, serde_json::Value>`, which does not fall into either of these cases.

Comment (Stranger6667) [reply jsonschema/src/output.rs]: Thanks!

**Review by Stranger6667 — APPROVED** [review]

**Comment by Stranger6667** [comment]
Released in 0.13.2!

**Pull request merged**